client/utils/utils.c

```diff
@@ -512,3 +512,23 @@ int validate_ascii(const char *str)
 	return 1;
 }
 
+size_t validate_ascii_safe(const char *str, size_t maxsize)
+{
+	if(str == NULL) {
+		return 0;
+	}
+
+	const char* start = str;
+	const char* stop = str + maxsize;
+
+	for(;start < stop;++str) {
+		if(*str < 32 || *str > 126) {
+			if(*str == '\0') {
+				return str - start;
+			}
+			return 0;
+		}
+	}
+
+	return maxsize;
```

Comment:
Why do you need to return the length?

Reply:
to allocate the memory for the string, yes in this case i could return just 0